packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -49,6 +49,15 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
     buttonVariant = VARIANT_MAPPING[variant];
   }
 
+  let hasIconOnlyChild = useMemo(() => {
+    if (React.Children.count(children) !== 1) {
+      return false;
+    }
+    let child = React.Children.toArray(children)[0];
+    // @ts-ignore
+    return React.isValidElement(child) && typeof child.type === 'function' && child.type().type.name === 'Icon';
```

Comment:
this is kinda ugly, so if someone has a better way of detecting an `Icon` type, definitely willing to update